third-party/re2/Makefile

```diff
@@ -21,6 +21,27 @@ CHPL_RE2_CXXFLAGS=-mmic
 endif
 endif
 
+
+# re2's makefile throws gcc specific flags that are not supported by the cray
+# compiler. This is intended to effectively replace the re2 makefile lines:
+#
+#     CXXFLAGS?=-Wall -O3 -g -pthread # can override
+#     RE2_CXXFLAGS?=-Wsign-compare -c -I. $(CCPCRE)  # required
+#     LDFLAGS?=-pthread
+#
+# with:
+#
+#     CXXFLAGS?= -O3
+#     RE2_CXXFLAGS?= -c -I. $(CCPCRE)
+#     LDFLAGS?=
+ifeq ($(CHPL_MAKE_TARGET_COMPILER),cray-prgenv-cray)
+CHPL_RE2_CXXFLAGS=-O3
```

Comment:
Note that I just copied this from the actual re2 makefile (Also I think -O3 is what we use in other places when OPTIMIZE=1, so I don't think it's out of place.). Ideally, I think we should just be passing in our CXXFLAGS and get the optimization level from that. I wasn't sure if there was a reason we weren't already doing that, so I was hoping @mppf might know.